cartographer_ros/cartographer_ros/offline_node_main.cc

```diff
@@ -162,7 +162,7 @@ void Run(const std::vector<string>& bag_filenames) {
   }
 
   for (const string& bag_filename : bag_filenames) {
-    if (sigint_triggered) {
+    if (!::ros::ok()) {
```

Comment:
I can't remember, but looking at the commit I can only hypothesize that our code does not like it if ROS is terminating while Cartographer is still running, so we handle Ctrl+C ourselves so that we always have `ros::ok()` until the very end?